File: src/main/kotlin/com/neko233/codec/orm/annotation/DeserializePostOrm.java
package com.neko233.codec.orm.annotation;


import com.neko233.codec.orm.DeserializeOrmFactoryApi;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * @author SolarisNeko
 * Date on 2023-12-01
 */
@Target(ElementType.FIELD)
@Retention(RetentionPolicy.RUNTIME)
public @interface DeserializePostOrm {

    // 来源字段名
    String fromFieldName();

    // 反序列化后的对象生产工厂类
    Class<? extends DeserializeOrmFactoryApi<?>> factoryClass();

    // 默认值, 若为空, 则 return null
    String defaultValue() default "";


}

File: src/main/kotlin/com/neko233/codec/orm/annotation/DeserializePostOrm.kt
package com.example.codec.postorm.annotation

import com.neko233.codec.orm.factory.DeserializeOrmFactoryApi
import kotlin.reflect.KClass

@Target(AnnotationTarget.FIELD)
@Retention(AnnotationRetention.RUNTIME)
annotation class DeserializePostOrm(
    // 来源字段名
    val fromFieldName: String = "",
    // 反序列化后的对象生产工厂类
    val factoryClass: KClass<out DeserializeOrmFactoryApi<*>>,
    // 默认值, 若为空, 则 return null
    val defaultValue: String = "",
)